src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,15 @@
 package core.basesyntax;
 
 public class MainApp {
+    private static void mail(String[] args) {
+        Machine truck = new Truck();
+        Machine bulldozer = new Bulldozer();
+        Machine excavator = new Excavator();
 
+        Machine[] vehicles = {truck, bulldozer, excavator};
```

Comment:
We introduce variables when
1) They are used more than once
2) When we need to add clarity to expression by giving it a name
3) When we need to split long complicated expression into parts
In other cases it's better to use value directly
```suggestion
        Machine[] vehicles = {new Truck(), new Bulldozer(), new Excavator()};
```